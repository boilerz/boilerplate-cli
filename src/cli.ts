#!/usr/bin/env node
import chalk from 'chalk';
import meow from 'meow';

import run, { Flags } from './commands/run';

const flags: Flags = {
  preferences: { type: 'boolean', alias: 'p' },
};

const cli = meow(
  `
  Usage
    $ boilerplate
  Options
    --preferences, -p  Setup preferences.
  Examples
    $ boilerplate -p
`,
  {
    flags,
  },
);

run(cli).catch((err: Error) => {
  console.log(chalk.bold.red(err.name), chalk.underline.red(err.message));
  console.log(chalk.gray(err.stack));
  process.exit(1);
});
