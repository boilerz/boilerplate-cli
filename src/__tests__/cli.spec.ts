import meow from 'meow';
import updateNotifier from 'update-notifier';

import pkg from '../../package.json';
import '../cli';

describe('cli', () => {
  it('should setup cli', () => {
    expect((meow as jest.Mock).mock.calls).toMatchSnapshot();
  });

  it('should call the update notifier', () => {
    expect(updateNotifier).toHaveBeenCalled();
    expect(updateNotifier({ pkg }).notify).toHaveBeenCalled();
  });
});
