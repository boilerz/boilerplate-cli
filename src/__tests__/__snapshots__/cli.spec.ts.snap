// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli should setup cli 1`] = `
Array [
  Array [
    "
  Usage
    $ boilerplate
  Options
    --preferences, -p  Setup preferences.
  Examples
    $ boilerplate -p
",
    Object {
      "flags": Object {
        "preferences": Object {
          "alias": "p",
          "type": "boolean",
        },
      },
    },
  ],
]
`;
